internal/kube/btp/service_test.go

```diff
@@ -0,0 +1,360 @@
+		done, err := readyFn(context.Background())
+		require.NoError(t, err)
+		require.True(t, done)
+	})
+
+	t.Run("failed binding", func(t *testing.T) {
+		scheme := runtime.NewScheme()
+		scheme.AddKnownTypes(GVRServiceBinding.GroupVersion())
+		givenBinding := fixUnstructuredServiceBinding()
+		givenBinding.Object["status"] = fixUnstructuredFailedCommonStatus()
+
+		dynamic := dynamic_fake.NewSimpleDynamicClient(scheme, givenBinding)
+		btpClient := NewClient(
+			dynamic,
+		)
+
+		readyFn := btpClient.IsBindingReady(context.Background(), "test-namespace", "test-name")
+		done, err := readyFn(context.Background())
+		require.Error(t, err)
+		require.False(t, done)
+	})
+
+	t.Run("binding not found", func(t *testing.T) {
+		scheme := runtime.NewScheme()
+		scheme.AddKnownTypes(GVRServiceBinding.GroupVersion())
+
+		dynamic := dynamic_fake.NewSimpleDynamicClient(scheme)
+		btpClient := NewClient(
+			dynamic,
+		)
+
+		readyFn := btpClient.IsBindingReady(context.Background(), "test-namespace", "test-name")
+		done, err := readyFn(context.Background())
+		require.ErrorContains(t, err, "servicebindings.services.cloud.sap.com \"test-name\" not found")
+		require.False(t, done)
+	})
+}
+
+func Test_btpClient_IsInstanceReadyFunc(t *testing.T) {
+	t.Run("ready instance", func(t *testing.T) {
+		scheme := runtime.NewScheme()
+		scheme.AddKnownTypes(GVRServiceInstance.GroupVersion())
+		givenInstance := fixUnstructuredServiceInstance()
+		givenInstance.Object["status"] = fixUnstructuredReadyCommonStatus()
+
+		dynamic := dynamic_fake.NewSimpleDynamicClient(scheme, givenInstance)
+		btpClient := NewClient(
+			dynamic,
+		)
+
+		readyFn := btpClient.IsInstanceReady(context.Background(), "test-namespace", "test-name")
+		done, err := readyFn(context.Background())
+		require.NoError(t, err)
+		require.True(t, done)
+	})
+
+	t.Run("failed instance", func(t *testing.T) {
+		scheme := runtime.NewScheme()
+		scheme.AddKnownTypes(GVRServiceInstance.GroupVersion())
+		givenInstance := fixUnstructuredServiceInstance()
+		givenInstance.Object["status"] = fixUnstructuredFailedCommonStatus()
+
+		dynamic := dynamic_fake.NewSimpleDynamicClient(scheme, givenInstance)
+		btpClient := NewClient(
+			dynamic,
+		)
+
+		readyFn := btpClient.IsInstanceReady(context.Background(), "test-namespace", "test-name")
+		done, err := readyFn(context.Background())
+		require.Error(t, err)
+		require.False(t, done)
+	})
+
+	t.Run("instance not found", func(t *testing.T) {
+		scheme := runtime.NewScheme()
+		scheme.AddKnownTypes(GVRServiceInstance.GroupVersion())
+
+		dynamic := dynamic_fake.NewSimpleDynamicClient(scheme)
+		btpClient := NewClient(
+			dynamic,
+		)
+
+		readyFn := btpClient.IsInstanceReady(context.Background(), "test-namespace", "test-name")
+		done, err := readyFn(context.Background())
+		require.ErrorContains(t, err, "serviceinstances.services.cloud.sap.com \"test-name\" not found")
+		require.False(t, done)
+	})
+}
+
+func fixServiceInstance() *ServiceInstance {
```

Comment:
Now we have corresponding methods `fix...` and `fixUnstructured...`. Both versions create very similar structures.
Is it possible to create only one structure and convert (to/from unstructured) it in second method?

Reply:
good point